src/tests/quantuminspire/qiskit/test_circuit_parser.py

```diff
@@ -197,11 +297,48 @@ def test_generate_cqasm_CorrectOutputRotationY(self):
         result = self._generate_cqasm_from_instructions(instructions, 2)
         self.assertTrue('Ry q[1], 0.654321\n' in result)
 
-    def test_generate_cqasm_CorrectOutputRotationZ(self):
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x3'}
+        result = self._generate_cqasm_from_instructions(instructions, 2)
+        self.assertTrue('not b[2,3]\nC-Ry b[0:3], q[1], 0.654321\nnot b[2,3]\n' in result)
+
+    def test_generate_cqasm_correct_output_rotation_z(self):
         instructions = [{'name': 'rz', 'qubits': [0], 'params': [np.pi / 2]}]
         result = self._generate_cqasm_from_instructions(instructions, 2)
         self.assertTrue('Rz q[0], 1.570796\n' in result)
 
         instructions = [{'name': 'rz', 'qubits': [1], 'params': [-np.pi / 2]}]
         result = self._generate_cqasm_from_instructions(instructions, 2)
         self.assertTrue('Rz q[1], -1.570796\n' in result)
+
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x1'}
+        result = self._generate_cqasm_from_instructions(instructions, 2)
+        self.assertTrue('not b[1,2,3]\nC-Rz b[0:3], q[1], -1.570796\nnot b[1,2,3]\n' in result)
+
+    def test_generate_cqasm_correct_output_unknown_gate(self):
+        instructions = [{'name': 'bla', 'qubits': [1], 'params': [-np.pi / 2]}]
+        self.assertRaisesRegex(ApiError, 'Gate bla not supported', self._generate_cqasm_from_instructions,
```

Comment:
It is about time that we start supporting the bla, and c-bla, gates.

Reply:
:) 👍
This bla-gate will swing the qubit to higher energy levels. But then we have the 'blabla' gate to test the non-existent gate and have to come up with another gate etc.